lib/index.js

```diff
@@ -172,16 +192,15 @@ module.exports = class DockerComposeParser {
 
     return validate(serviceSchema, rawService)
     // Map raw properties
-    .then(() => {
-      rawService.environment = Mapper.mapKeyValueObjectToArray(rawService.environment)
-      return rawService
-    })
-    // Populate Warnings
-    .then(rawService => Warning.unsupportedKeys(rawService, warnings, serviceSchemaKeys))
-    .then(rawService => Warning.buildAndImage(rawService, warnings))
-    // Populate properties
-    .then(rawService => DockerComposeParser._mapRawServiceToService(service, rawService, repositoryName, warnings))
-    .then(service => DockerComposeParser._removeUnusedKeys(service, warnings))
+      .then(() => {
```

Comment:
That was actually webstorm that did that.  But yeah, your decisions are also terrible, and you should feel bad